internal/app/server/handler_test.go

```diff
@@ -88,6 +89,105 @@ func TestNewParcel(t *testing.T) {
 	}
 }
 
+func TestGetParcel(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	parcel := model.Parcel{
+		UserID:             1,
+		SourceAddress:      "Dhaka Bangladesh",
+		DestinationAddress: "Pabna Shadar",
+		SourceTime:         time.Date(2020, time.April, 11, 21, 34, 01, 0, time.UTC),
+		ParcelType:         "Document",
+		Price:              200.0,
+		CarrierFee:         180.0,
+		CompanyFee:         20.0,
+		CreatedAt:          time.Date(2020, time.April, 11, 21, 34, 01, 0, time.UTC),
+		UpdatedAt:          time.Date(2020, time.April, 11, 21, 34, 01, 0, time.UTC),
+	}
+
+	testCases := []struct {
+		desc          string
+		mockParcelSvc func() *mock_service.MockParcelService
+		parcelID      string
+		expStatusCode int
+		expResponse   string
+	}{
+		{
+			desc: "should success",
+			mockParcelSvc: func() *mock_service.MockParcelService {
+				s := mock_service.NewMockParcelService(ctrl)
+				s.EXPECT().GetParcelByID(gomock.Any(), 1).Return(parcel, nil)
+				return s
+			},
+			parcelID:      "1",
+			expStatusCode: http.StatusOK,
+			expResponse:   `{"success":true,"errors":null,"data":{"id":0,"user_id":1,"carrier_id":0,"status":0,"source_address":"Dhaka Bangladesh","destination_address":"Pabna Shadar","source_time":"2020-04-11T21:34:01Z","type":"Document","price":200,"carrier_fee":180,"company_fee":20,"created_at":"2020-04-11T21:34:01Z","updated_at":"2020-04-11T21:34:01Z"}}`,
+		},
+		{
+			desc: "should return ID not exist",
+			mockParcelSvc: func() *mock_service.MockParcelService {
+				s := mock_service.NewMockParcelService(ctrl)
+				s.EXPECT().GetParcelByID(gomock.Any(), 1).Return(model.Parcel{}, model.ErrInvalid)
+				return s
+			},
+			parcelID:      "1",
+			expStatusCode: http.StatusBadRequest,
+			expResponse:   `{"success":false,"errors":[{"code":"INVALID","message":"invalid","message_title":"This ID does not exist.","severity":"error"}],"data":null}`,
+		},
+		{
+			desc: "should return internal server error",
+			mockParcelSvc: func() *mock_service.MockParcelService {
+				s := mock_service.NewMockParcelService(ctrl)
+				s.EXPECT().GetParcelByID(gomock.Any(), 1).
+					Return(model.Parcel{}, errors.New("server-error"))
+				return s
+			},
+			parcelID:      "1",
+			expStatusCode: http.StatusInternalServerError,
+			expResponse:   `{"success":false,"errors":[{"code":"SERVER_ERROR","message":"server-error","message_title":"Failed to fetch parcel 1","severity":"error"}],"data":null}`,
+		},
+		{
+			desc: "should return internal server error",
+			mockParcelSvc: func() *mock_service.MockParcelService {
+				s := mock_service.NewMockParcelService(ctrl)
+				s.EXPECT().GetParcelByID(gomock.Any(), 1).
+					Return(model.Parcel{}, model.ErrNotFound)
+				return s
+			},
+			parcelID:      "1",
+			expStatusCode: http.StatusBadRequest,
```

Comment:
it should be not found error not bad request

Reply:
not my code though I have fixed on mine.

Reply:
it comes with your PR so definitely its your code

Reply:
It comes after merging with master so don't be confused with that.

Reply:
i am not confused. any code changed by your PR its your code. if it was in master then. it should not show change

Reply:
If there is something wrong on master or other branch that we didn't notice before, we should correct them whenever it is seen.